src/provider/interface.ts

```diff
@@ -34,7 +34,7 @@ import type {
 } from '../types';
 
 export abstract class ProviderInterface {
-  public abstract channel: RpcChannel;
+  public abstract channel: RpcChannel | undefined;
```

Comment:
This change doesn't seem necessary because of the `@ts-expect-error` in `src/provider/rpc.ts` on line 44.

Reply:
I wasn't sure will ti effect external implementations